src/types/types/deep-keys.ts

```
@@ -0,0 +1,12 @@
+/**
+ * Convert an object to a string of its keys (Same as keyof but can go deeper)
+ */
+export type DeepKeys<T> = T extends object
+	? {
+			[K in keyof T]-?: K extends string
+				? T[K] extends Date | Array<unknown> | Function
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)